File: src/main/java/com/procurement/evaluation/model/dto/UpdateAwardRequestDto.java
package com.procurement.evaluation.model.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.procurement.evaluation.model.dto.ocds.Award;
import javax.validation.Valid;
import javax.validation.constraints.NotNull;
import lombok.Getter;

@Getter
public class UpdateAwardRequestDto {

    @Valid
    @NotNull
    @JsonProperty("award")
    private final Award award;

    @JsonCreator
    public UpdateAwardRequestDto(@JsonProperty("award") final Award award) {
        this.award = award;
    }
}

File: src/main/kotlin/com/procurement/evaluation/model/dto/UpdateAwardRequestDto.kt
package com.procurement.evaluation.model.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.procurement.evaluation.model.dto.ocds.Document
import com.procurement.evaluation.model.dto.ocds.Status
import javax.validation.Valid
import javax.validation.constraints.NotNull

data class UpdateAwardRequestDto @JsonCreator constructor(

        @field:Valid @field:NotNull
        val award: AwardUpdate
)

@JsonInclude(JsonInclude.Include.NON_NULL)
data class AwardUpdate @JsonCreator constructor(

        @field:NotNull
        var statusDetails: Status,

        var description: String?,

        var documents: List<Document>?
)